### PR TITLE
rename global to core

### PR DESCRIPTION
**Pull request by Sertonix**
This renames the main global to `core`. The old `atom` global is still available. It worked for me without problems.
This is rather a suggestion for the future than a quick fix but I just wanted to show that it works.

To find the occurrences I used the search in files function with these options:
- match case
- use regex: `(?<![a-zA-Z]_"'/\-:)atom(?![a-zA-Z_"'/\-:]|\.io)`
- files pattern: `!node_modules,!ppm/lib,*.js,*.coffee`

There are multiple locations where both `core` and `atom` need to be defined:
- [src/initialize-application-window.js#L71](https://github.com/sertonix/pulsar/blob/core-global/src/initialize-application-window.js#L71)
- [src/initialize-benchmark-window.js#L82](https://github.com/sertonix/pulsar/blob/core-global/src/initialize-benchmark-window.js#L82) (Removed in master)
- [spec/main-process/atom-application.test.js#L799](https://github.com/sertonix/pulsar/blob/core-global/spec/main-process/atom-application.test.js#L799)
- [spec/jasmine-test-runner.js#L63](https://github.com/sertonix/pulsar/blob/core-global/spec/jasmine-test-runner.js#L63)

Error difference to the master branch:
- +3 failed tests in `test development helpers packages`
- error in `test editor related packages`

### CODE REVIEW

File: packages/about/spec/about-status-bar-spec.js [hunk 1]
```diff
@@ -14,27 +14,27 @@ describe('the status bar', () => {
     spyOn(window.localStorage, 'getItem').andCallFake(key => {
       return storage[key];
     });
-    spyOn(atom, 'getVersion').andCallFake(() => {
-      return atomVersion;
```
Comment (Sertonix) [inline packages/about/spec/about-status-bar-spec.js:18]: TODO revert rename!

File: spec/window-event-handler-spec.js [hunk 2]
```diff
@@ -6,25 +6,25 @@ describe('WindowEventHandler', () => {
...
+    const initialPath = core.project.getPaths()[0];
+    spyOn(core, 'getLoadSettings').andCallFake(() => {
+      const loadSettings = core.getLoadSettings.originalValue.call(core);
       loadSettings.initialPath = initialPath;
       return loadSettings;
     });
-    atom.project.destroy();
+    core.project.destroy();
     windowEventHandler = new WindowEventHandler({
-      atomEnvironment: atom,
```
Comment (Sertonix) [inline spec/window-event-handler-spec.js:19]: TODO revert rename

**Comment by confused-Techie** [comment]
I get the want to make this change, but this change would have to thoroughly reviewed, since the likely-hood of breaking massive amounts of ours, and others codebase is very high.

Now it might be important to not the ramifications of this change and the recommendation of never doing this is currently documented in our [Contributing Docs](https://github.com/pulsar-edit/.github/blob/main/project-birth/CONTRIBUTING-DURING-START.md#rebranding-in-code), although there are more talks about making this change [over on our discussions](https://github.com/orgs/pulsar-edit/discussions/28#discussioncomment-3954592), which I'm assuming is what prompted this PR.

So it should be reviewed by @pulsar-edit/core, but this change shouldn't be taken likely and may take some time to be merged. Since if we do make this change we *need* to have an alternative, in my opinion, available right away, so that massive amounts of code doesn't instantly break.

**Comment by mauricioszabo** [comment]
This change is really risky with absolutely no gain. Why, exactly?

**Comment by Daeraxa** [comment]
> This change is really risky with absolutely no gain. Why, exactly?

I believe it was related to a conversation about having a generic "engine" name so that any future forks of the project wouldn't have to go and rebrand a ton of stuff and it could be done in one generic place.

**Comment by Sertonix** [comment]
As mentioned above the idea is to make the program more generic. Since refactoring a lot of stuff can easily cause errors it would be possible to only define the generic global but don't use it for the time being. Only the 4 places mentioned in my description would need to be changed. If that is ok I would create a new pull request for that.

**Comment by ThatXliner** [comment]
Can't we just alias it and return a deprecation warning if they use the atom global?

**Comment by Sertonix** [comment]
That is what I meant. But I would not (yet) use a deprecated warning because it would spam everything

**Comment by Sertonix** [comment]
Closing this in favor to #113 but I would suggest keeping the discussion here.

**Pull request closed without merging**